data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpointsTest.java

```diff
@@ -22,20 +22,22 @@
 import java.util.Optional;
 import org.junit.jupiter.api.Test;
 import tech.pegasys.teku.api.response.v1.beacon.GetStateFinalityCheckpointsResponse;
-import tech.pegasys.teku.api.schema.BeaconState;
+import tech.pegasys.teku.api.schema.Checkpoint;
 import tech.pegasys.teku.beaconrestapi.AbstractBeaconHandlerTest;
+import tech.pegasys.teku.datastructures.state.BeaconState;
 import tech.pegasys.teku.datastructures.util.DataStructureUtil;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 
 public class GetStateFinalityCheckpointsTest extends AbstractBeaconHandlerTest {
   final DataStructureUtil dataStructureUtil = new DataStructureUtil();
-  final BeaconState state = new BeaconState(dataStructureUtil.randomBeaconState());
+  final BeaconState state = dataStructureUtil.randomBeaconState();
```

Comment:
this was wrapped because the tests in the API were attempting to use the api level objects (the api.schema.BeaconState)
is there a reason we need the internal object?

Reply:
we were unnecessarily serializing the whole beacon state object to only get the checkpoint information. seemed like wasted CPU cycles.

Reply:
its more that the api level then may have dependencies on internal structures, to be honest i'd far prefer its only using api level objects unless explicitly needing the internal ones for a test...

Reply:
But we still get the internal beacon state object in ChainDataProvider. So are we trying to keep the internal only there?

Reply:
I switched to making the `FinalityCheckpointsResponse` object in ChainDataProvider so that the rest API handler does not touch the internal object.